joomla/joomla-cms
# Text::script should update existing script strings

Pull Request for Issue #41516 .

### Summary of Changes

The `HTMLDocument::setHeadData` restores `scriptOptions` correctly, hovewer any further call of `Text::script` override previously restored strings.



### Testing Instructions
Please follow #41516 .
Or add :
```php
Text::script('JYES');
$this->document->addScriptOptions('yes', 'Yes');
$this->document->getWebAssetManager()
    ->addInlineScript('console.log(Joomla.getOptions("yes", "nope") + " " + Joomla.Text._("JYES", "Nope"))', [], ['type' => 'module']);
```
After this line https://github.com/joomla/joomla-cms/blob/1e7527b60ec42032abe7bfb9c000bacf38f65845/components/com_content/src/View/Article/HtmlView.php#L242
Enable "Conservative caching"
Open any Article view.
Check browser console.


### Actual result BEFORE applying this Pull Request
First request: `Yes Yes`
All following requests: `Yes Nope`


### Expected result AFTER applying this Pull Request
Every reques there a log `Yes Yes`


### Link to documentations
Please select:
